src/components/Icon/Icon.tsx

```diff
@@ -36,12 +38,13 @@ type Props = {
  * @param name One of built-in Vanilla icons or a name of a custom icon that follows `p-icon--{name}` convention.
  * @returns Icon
  */
-const Icon = ({ name }: Props): JSX.Element => (
-  <i className={`p-icon--${name}`} />
+const Icon = ({ className, name }: Props): JSX.Element => (
+  <i className={classNames(className, `p-icon--${name}`)} />
```

Comment:
Is calling this `className` a bit misleading given this only applies to `p-icon` styles?